api/resources/gene_information.py

```diff
@@ -135,6 +148,204 @@ def get(self, species="", gene_id=""):
+                if chromosome == 'Chr1':
+                    chromosomeId = '1'
+                elif chromosome == 'Chr2':
+                    chromosomeId = '2'
+                elif chromosome == 'Chr3':
+                    chromosomeId = '3'
+                elif chromosome == 'Chr4':
+                    chromosomeId = '4'
+                elif chromosome == 'Chr5':
+                    chromosomeId = '5'
+                elif chromosome == 'ChrC':
+                    chromosomeId = 'C'
+                elif chromosome == 'ChrM':
+                    chromosomeId = 'M'
+                else:
+                    return BARUtils.error_exit("Invalid chromosome"), 400
+            else:
+                return BARUtils.error_exit("No data for the given species"), 400
+
+            # Construct the query
+            atnumg = "AT" + chromosomeId + "G"
+            query1 = db.select(database.geneId, database.Start, database.End, database.Strand).where(
+                database.Type == "gene",
+                database.geneId.startswith(atnumg),
+                (
+                    database.Start.between(startParam, endParam) |
+                    database.End.between(startParam, endParam) |
+                    ((database.Start < startParam) & (database.End > endParam))
+                )
+            )
+            result1 = db.session.execute(query1).all()
+            gene_ids = [row[0] for row in result1]
+
+            # Get aliases
+            query2 = db.select(EPlant2AgiAlias.agi, EPlant2AgiAlias.alias).where(EPlant2AgiAlias.agi.in_(gene_ids))
+            result2 = db.session.execute(query2).all()
+            all_aliases = {}
+            for row in result2:
+                if row[0] not in all_aliases:
+                    all_aliases[row[0]] = []
+                all_aliases[row[0]].append(row[1])
+
+            # Get annotation
+            query3 = db.select(EPlant2AgiAnnotation.agi, EPlant2AgiAnnotation.annotation).where(EPlant2AgiAnnotation.agi.in_(gene_ids))
+            result3 = db.session.execute(query3).all()
+            all_annotations = {}
+            for row in result3:
+                temp = row[1].split('__')
+                if len(temp) > 1:
+                    all_annotations[row[0].upper()] = temp[1]
+                else:
+                    all_annotations[row[0].upper()] = temp[0]
+
+            genes = []
+            for row in result1:
+                gene = {
+                    'id': row[0],
+                    'start': row[1],
+                    'end': row[2],
+                    'strand': row[3],
+                    'aliases': all_aliases.get(row[0], []),
+                    'annotation': all_annotations.get(row[0], None)
+                }
+
+                genes.append(gene)
+            return BARUtils.success_exit(genes)
+
+        except Exception as e:
+            return BARUtils.error_exit(str(e)), 400
+
+
+@gene_information.route("/gene_query/<string:species>")
```

Comment:
Rewrite this to HATEOAS style.

Reply:
@TeresasaZ  @asherpasha  You can just refactor this into a POST request instead of a GET with URL params.